### ISSUE TITLE
[AuthenticationService] type 'String' is not a subtype of type 'User'

### ISSUE THREAD

**Issue by FredvanRijswijk** [FilledStacks/flutter-tutorials#43]
```
Future _populateCurrentUser(FirebaseUser user) async {
    //_firebaseAuth.signOut();
    if (user != null) {
      _currentUser = await _firestoreService.getUser(user.uid);
      await _analyticsService.setUserProperties(
        userId: user.uid,
        userRole: _currentUser.userRole,
      );
    }
  }
```

"_currentUser 
 Exception has occurred.
_TypeError (type 'String' is not a subtype of type 'User')"

**Comment by FilledStacks** [comment]
That's  when the user fetch fails. you should handle it differently.

Check the result first. if it's a string show a dialog message with the result as the message, if it's a user set the result to _currentUser.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by thanooshan** [comment]
Hi all, I also got the exact same error. Then I printed out the error message in `getUser()` in `FirstoreService`. I got `FIRFirestoreErrorDomain` as the error message. Then I searched regarding this error, then found out it's related to `Firebase permission`. 

Then made a small modification in the `Firebase rules` section.

```
match /{document=**} {      
    allow read: if loggedIn(); //previously it was TRUE. I updated to loggedIn()
    allow write: if false;
}
```
Then the error was gone. I don't whether this is good. but it's works for now. 

@FilledStacks The change I've made correct or wrong? Is there any other way to do it?
